brave-tests/src/test/java/brave/ThreadLocalSpanClassLoaderTest.java

```diff
@@ -0,0 +1,49 @@
+package brave;
+
+import brave.propagation.ThreadLocalSpan;
+import org.junit.Test;
+import zipkin2.reporter.Reporter;
+
+import static brave.test.util.ClassLoaders.assertRunIsUnloadable;
+
+public class ThreadLocalSpanClassLoaderTest {
+
+  @Test public void noop_unloadable() {
+    assertRunIsUnloadable(getClass().getClassLoader(), CurrentTracerUnassigned.class);
+  }
+
+  static class CurrentTracerUnassigned implements Runnable {
+    @Override public void run() {
+      ThreadLocalSpan.CURRENT_TRACER.next();
+    }
+  }
+
+  @Test public void nextAndRemove_unloadable() {
+    assertRunIsUnloadable(getClass().getClassLoader(), CurrentTracerRemovesSpan.class);
+  }
+
+  static class CurrentTracerRemovesSpan implements Runnable {
+    @Override public void run() {
+      try (Tracing tracing = Tracing.newBuilder().build()) {
+        ThreadLocalSpan.CURRENT_TRACER.next();
```

Comment:
this test bombs.. I'm not sure I have the debugging skills to figure out what is holding up the classloader. General tricks suggested if folks know some libraries or tools that can print out what's holding something from being gc'd. cc @raphw